test/e2e/lifecycle/kubelet_security.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package lifecycle
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"time"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	// Try all the proxy tests this many times (to catch even rare flakes).
+	proxyAttempts = 10
+	// Only print this many characters of the response (to keep the logs
+	// legible).
+	maxDisplayBodyLen = 100
+
+	// We have seen one of these calls take just over 15 seconds, so putting this at 30.
+	proxyHTTPCallTimeout = 30 * time.Second
+)
+
+var _ = SIGDescribe("Ports Security Check [Feature:KubeletSecurity]", func() {
+	f := framework.NewDefaultFramework("kubelet-security")
+
+	prefix := "/api/v1"
+	// make sure kubelet readonly (10255) and cadvisor (4194) ports are disabled via API server proxy
+	It("should not be able to proxy to the readonly kubelet port 10255 using proxy subresource", func() { portProxyDisabledTest(f, prefix+"/nodes/", ":10255/pods/") })
+	It("should not be able to proxy to cadvisor port 4194 using proxy subresource", func() { portProxyDisabledTest(f, prefix+"/nodes/", ":4194/proxy/containers/") })
+
+	// make sure kubelet readonly (10255) and cadvisor (4194) ports are closed on the public IP address
+	disabledPorts := []int{10255, 4194}
+	for _, port := range disabledPorts {
+		It(fmt.Sprintf("should not have port %d open on its public IP address", port), func() { portClosedTest(f, port) })
+	}
+})
+
+func doProxy(f *framework.Framework, path string, i int) (body []byte, statusCode int, d time.Duration, err error) {
+	// About all of the proxy accesses in this file:
+	// * AbsPath is used because it preserves the trailing '/'.
+	// * Do().Raw() is used (instead of DoRaw()) because it will turn an
+	//   error from apiserver proxy into an actual error, and there is no
+	//   chance of the things we are talking to being confused for an error
+	//   that apiserver would have emitted.
+	start := time.Now()
+	body, err = f.ClientSet.CoreV1().RESTClient().Get().AbsPath(path).Do().StatusCode(&statusCode).Raw()
+	d = time.Since(start)
+	if len(body) > 0 {
+		framework.Logf("(%v) %v: %s (%v; %v)", i, path, truncate(body, maxDisplayBodyLen), statusCode, d)
+	} else {
+		framework.Logf("%v: %s (%v; %v)", path, "no body", statusCode, d)
+	}
+	return
+}
```

Comment:
See [NodeProxyRequest](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/util.go#L4497) for a better approach (or maybe refactor that to have a version that allows you to specify the port)